_posts/2015-08-28-gulp-workflow.md

```diff
@@ -0,0 +1,119 @@
+---
+layout: post
+title: "Automate Your Tasks Easily with Gulp"
+author: nirajan_basnet
+modified: 2015-08-28
+tags: [ gulp, javascript ]
+---
+
+# Gulp Workflow
+
+![alt text](https://d262ilb51hltx0.cloudfront.net/max/1400/1*jbeaev-Xdy4S4pNNJHwA0Q.png "Logo Title Text 1")
+
+It was the first time I was using Gulp. The experience, I must say, was quite enjoyable.
+[Gulp](https://www.gulpjs.com) is a javascript automated task runner. And it prefers code over configuration and gives you the plugins to create tasks that fit your specific needs and helps in the production process.
+
+***
+
+### Why we need Gulp ?
+As front-end developers, we often need to look at the tools we use and decide if we are using the right tool for the job. It creates a good practice within the flow to the system. You’ll save hours of effort and reduce human error. It’s a build tool that helps to speed up your workflow by :
+* Repetitive automating tasks
+* Optimising Images
+* Concatenating
+* Minifying Css and Javascript
+* Browser Prefixes
+
+It focuses on piping the result of one task into the next until you reach the point where you’ve completed all your tasks on that particular file.
+To do a front-end project, this was a new world to me! Initially, I usually just dragged my project folder into Prepros and wrote some Scss or LESS but it makes platform dependent which is very difficult to share and co-operate with other members. But after using Gulp the whole thing changed. My workflow became clean, effective and the entire process helped me save a lot of time. So, here’s the further process of how we can collaborate Gulp to our projects.

```

Comment:
`nut it makes the process platform dependent`

`how we can include Gulp to our projects`